### DISCUSSION TITLE
Anomalib Calculating Pixel Metrics

### DISCUSSION THREAD

**Discussion by yxteo2** [open-edge-platform/anomalib#2536] [Feature Requests]
Hi guys,

I am trying to calculate the pixel AUROC for the images, since now I am importing image path to predict the contour.
I am using the AUROC function from anomalib.metrics.AUROC, can anyone tell me if I am doing it in the right way? 
I feel something is off

for img_path in image_list_path:
            try:
                
                full_img_path = img_path
                print(full_img_path)

                img_original1 = cv2.imread(full_img_path)
                if img_original1 is None:
                    raise FileNotFoundError(f"Image not found at {full_img_path}")

                    image_blob = cv2.dnn.blobFromImage(img_original1, scalefactor=1 / 255.0, size=self.image_size, swapRB=True)
                    as_tensor=True
                    if img_original1 is not None:
                        img_original = cv2.cvtColor(img_original1, cv2.COLOR_BGR2RGB)  # Convert to RGB format
                    else:
                        raise FileNotFoundError(f"Image not found at the specified path: {img_path}")
                 image = to_dtype(to_image(img_original), torch.float32, scale=True) if as_tensor else np.array(img_original) / 255.0
                 self.image = image
                 predictions, unnormalize_anomaly_map, unnormalize_pred_score = self.run_onnx_inference(image, image_blob)

                
                if self.mask_folder is not None:
                    try:
                        # Assuming unnormalized anomaly map and ground truth mask (same shapes)
                        mask_full_path = full_img_path.replace('test', 'ground_truth')
                        mask_full_path = mask_full_path.split('.')[0] + '_mask.' + mask_full_path.split('.')[1]
                        
                        if full_img_path.split('\\')[-2] == 'good':
                            # Create a black mask map for 'good' images
                            mask_map = np.zeros_like(unnormalize_anomaly_map, dtype=np.uint8)
                        else:
                            # Load the mask map
                            mask_map = cv2.imread(mask_full_path, cv2.IMREAD_GRAYSCALE)
                            if mask_map is None:
                                raise FileNotFoundError(f"Mask not found at {mask_full_path}")

                        # Resize the mask map to match the anomaly map dimensions
                        mask_map = cv2.resize(mask_map, (unnormalize_anomaly_map.shape[1], unnormalize_anomaly_map.shape[0]))

                        # Normalize predictions (anomaly map) to [0, 1]
                        preds = (unnormalize_anomaly_map / 255.0).astype(np.float32)

                        # Ensure the mask is binary and convert to integer
                        mask_map = (mask_map > 127).astype(np.int64)  # Thresholding and conversion to int64

                        # Convert predictions and targets to PyTorch tensors
                        preds_tensor = torch.tensor(preds.flatten(), dtype=torch.float32)
                        targets_tensor = torch.tensor(mask_map.flatten(), dtype=torch.int64)  # Ensure int64 dtype for AUROC

                        # Initialize and compute the AUROC metric
                        auroc_metric = AUROC()
                        auroc_metric.update(preds_tensor, targets_tensor)
                        auroc_score = auroc_metric.compute()

                        print(f"Pixel-wise AUROC Score: {auroc_score:.4f}")
